Sources/MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -187,3 +187,18 @@ extension String {
         return "MB" + self
     }
 }
+
+extension MeasurementSystem {
+    /// Converts `MapboxDirections.MeasurementSystem` into `LengthFormatter.Unit`.
+    public var distanceUnit: LengthFormatter.Unit {
+        return self == .metric ? .kilometer : .mile
+    }
+}
+
+extension LengthFormatter.Unit {
+    /// Converts `LengthFormatter.Unit` into `MapboxDirections.MeasurementSystem`.
+    public var measurementSystem: MeasurementSystem {
```

Comment:
Replace this with a `MeasurementSystem` initializer that takes a `LengthFormatter.Unit` as an unlabeled argument (and vice versa).